views/urls.go

```diff
@@ -3,15 +3,63 @@ package views
 import (
 	"net/url"
 
+	"github.com/a-h/templ"
 	"github.com/ugent-library/bind"
 )
 
-func urlWithQuery(u *url.URL, v any) *url.URL {
-	vals, err := bind.EncodeQuery(v)
+type URLBuilder struct {
+	url *url.URL
+}
+
+func NewURLBuilder(base string) *URLBuilder {
+	url, err := url.Parse(base)
+	if err != nil {
+		// TODO: how to deal with this
+		panic(err)
+	}
+
+	return &URLBuilder{url}
+}
+
+func (builder *URLBuilder) WithPath(path ...string) *URLBuilder {
+	builder.url = builder.url.JoinPath(path...)
+
+	return builder
+}
+
+func (builder *URLBuilder) AddQuery(key string, value string) *URLBuilder {
+	query := builder.url.Query()
+	query.Add(key, value)
+
+	builder.url.RawQuery = query.Encode()
+
+	return builder
+}
+
+func (builder *URLBuilder) WithQuery(query any) *URLBuilder {
+	vals, err := bind.EncodeQuery(query)
 	if err != nil {
-		return u
+		return builder
 	}
-	newU := *u
-	newU.RawQuery = vals.Encode()
-	return &newU
+
+	builder.url.RawQuery = vals.Encode()
+	return builder
+}
```

Comment:
This is intended for use with a bindable struct (using `` `query:"param"` `` descriptors). Not sure if we should validate that the argument is indeed a bindable struct. If you pass a string eg, that would result in a weird query string:

```go
NewURLBuilder("https://example.com").WithQuery("foo=bar").String()

// becomes: https://example.com?=foo%3Dbar
```

Reply:
@koen Add a separate method `QueryString` or `RawQuery`?

Reply:
Done